pkg/apis/mattermost/v1alpha1/clusterinstallation_utils.go

```diff
@@ -62,10 +62,28 @@ func (mattermost *ClusterInstallation) SetDefaults() error {
 	mattermost.Spec.Minio.SetDefaults()
 	mattermost.Spec.Database.SetDefaults()
 	err := mattermost.Spec.BlueGreen.SetDefaults(mattermost)
```

Comment:
We need to check if this error is nil or not before we overwrite it on the next line.